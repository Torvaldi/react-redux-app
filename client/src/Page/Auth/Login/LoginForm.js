import React, { Component } from 'react';
import { connect } from 'react-redux';
import { login, changeUsername, changePassword, resetErrorLogin, resetSucessRegister } from 'Page/Auth/action';
import { Link } from 'react-router-dom';
import propTypes from 'prop-types';

import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Alert from 'components/Alerte/Alert';
import LoadingPage from 'components/LoadingPage/LoadingPage';
import '../style.css';
import { withRouter, Redirect } from 'react-router-dom';


const mapStateToProps = (state) => ({ ...state.auth });

const mapDispatchToProps = (dispatch) => ({
  onChangeUsername: (username) =>
    dispatch(changeUsername(username)),
  onChangePassword: (password) =>
    dispatch(changePassword(password)),
  onSubmit: (username, password) =>
    dispatch(login({ username, password })),
  onResetErrorLogin: () =>
    dispatch(resetErrorLogin()),
  onResetMessageRegister: () =>
    dispatch(resetSucessRegister()),
});

class LoginForm extends Component {

  changeUsername = (event) => this.props.onChangeUsername(event.target.value);
  changePassword = (event) => this.props.onChangePassword(event.target.value);

  /**
   * @param {string} username
   * @param {string} password
   */
  submitForm = (username, password) => (event) => {
    event.preventDefault();
    if (username && password) {
      this.props.onSubmit(username, password);
    }
  }

  /**
   * @param {string} errorLogin
   * Reset errorLogin state if the user click on an input
   */
  resetError = (errorLogin) => (event) => {
    event.preventDefault();
    if (errorLogin) {
      this.props.onResetErrorLogin();
    }
  }

  /**
   * @param {string}
   */
  saveToken = (token) => {
    document.cookie = "token=" + token;
  }

  /**
   * @param {array}
   * @return DOM
   */
  printErrorLogin = (errorLogin) => {
    return(
      errorLogin.map((error, index) => {
        return <Alert message={error} type={0} key={index} />
      })
    );
  }

  render() {
    const { username, password, errorLogin, token, isLoginLoading } = this.props;
    
    if (token !== null && errorLogin === null) {
      this.saveToken(token);
      return <Redirect to="/game" />
      
    } else {

      let alerteLogin = "";
      if (errorLogin) {
        alerteLogin = this.printErrorLogin(errorLogin);
      }
      return (
        <form onSubmit={this.submitForm(username, password)}>
          {alerteLogin}
          <div className="formField">
            <TextField
              id="username"
              label="Username"
              placeholder="Username"
              className="textField"
              onChange={this.changeUsername}
              onClick={this.resetError(errorLogin)}
              margin="normal"
              variant="filled"
              required
            />
          </div>
          <div className="formField">
            <TextField
              id="Password"
              label="Password"
              type="password"
              className="textField"
              autoComplete="current-password"
              onChange={this.changePassword}
              onClick={this.resetError(errorLogin)}
              margin="normal"
              variant="filled"
              required
            />
          </div>
          <div className="buttonField">
            <Button className="buttonLogin" type="submit" size="large" variant="contained" color="secondary">
              Login
            </Button>
            <Link to="/register">
              <Button className="buttonRegister" size="large" variant="outlined" color="secondary">
                Sign up
              </Button>
            </Link>
          </div>
          {isLoginLoading === true ? <LoadingPage overlay={true} ></LoadingPage> : ''}
        </form>
      );

    }


  }
}

LoginForm.propTypes = {
  onChangeUsername: propTypes.func,
  onChangePassword: propTypes.func,
  onSubmit: propTypes.func,
  onResetError: propTypes.func,
  username: propTypes.string,
  password: propTypes.string,
  errorLogin: propTypes.array,
}


let component = connect(mapStateToProps, mapDispatchToProps)(LoginForm);
export default withRouter(component);