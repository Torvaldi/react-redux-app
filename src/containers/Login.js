import React from 'react';
import './../css/login.css';
import { connect } from 'react-redux';

import { UPDATE_FIELD_AUTH, LOGIN } from './../actions';
import { login } from './../actions/auth';

const mapStateToProps = (state) => ({ ...state.auth });

const mapDispatchToProps = (dispatch) => ({
  onChangeUsername :  (value) =>
    dispatch({type: UPDATE_FIELD_AUTH, key: 'username', value}),
  onChangePassword :  (value) =>
    dispatch({type: UPDATE_FIELD_AUTH, key: 'password', value}),
  onSubmit: (username, password) =>
    login(dispatch, {username, password}),
});


class Login extends React.Component {

  constructor(){
    super();
    this.changeUsername = (event) => this.props.onChangeUsername(event.target.value);
    this.changePassword = (event) => this.props.onChangePassword(event.target.value);
    this.submitForm = (username, password) => (event) => {
      event.preventDefault();
      this.props.onSubmit(username, password);
    }
  }

  render(){
    const { username, password } = this.props;
    return (
      <section className="auth-layout">
        <article>
          <form onSubmit={this.submitForm(username, password)}>
            <label htmlFor="">Username</label>
            <input 
              onChange={this.changeUsername} 
              type="text"/>
            <label htmlFor="">Password</label>
            <input 
              onChange={this.changePassword} 
              type="text"/>
            <input type="submit"/>
          </form>
          <span>{this.props.error}</span>
          <span>{this.props.token}</span>
        </article>
        <article>
          Image
        </article>
      </section>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);
