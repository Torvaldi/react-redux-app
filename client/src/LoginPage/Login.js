import React, { Component }from 'react';

import LoginForm from './LoginForm';
import Image from '../components/LoginImage/LoginImage';
import AuthLayout from '../components/AuthLayout/AuthLayout';
import image from '../asset/login.jpg';
import { isLogIn } from './../helper/auth';
import { withRouter } from 'react-router-dom';

class Login extends Component {

  render(){

    // prevent from accessing the page if the user is already log in
    isLogIn().then((response) => {

    if(response === false) return
    
    this.props.history.push('/game');

    });

    return (
        <AuthLayout
        text={
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vesti"
        }
        left={
          <LoginForm />
        } 
        right={
          <Image image={image} />
        }
        />
    );
}
  }
  

export default withRouter(Login)
