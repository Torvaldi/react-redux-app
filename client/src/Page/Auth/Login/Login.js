import React, { Component }from 'react';

import LoginForm from './LoginForm';
import Image from '../../../components/LoginImage/LoginImage';
import AuthLayout from '../../../components/AuthLayout/AuthLayout';
import image from '../../../asset/login.jpg';
import { isLogIn } from '../../../helper/auth';
import { withRouter, Redirect } from 'react-router-dom';

class Login extends Component {

  render(){

    // prevent from accessing the page if the user is already log in
    isLogIn().then((response) => {

      if(response === false) return
      
      return <Redirect to="/game" />

    });

    return (
        <AuthLayout
        text={
          "Login / Sign-in"
        }
        left={
          <LoginForm />
        } 
        right={
          <Image image={image} />
        }
        />
    );
}
  }
  

export default withRouter(Login)
