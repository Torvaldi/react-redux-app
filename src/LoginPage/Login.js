import React from 'react';

import LoginForm from './LoginForm';
import Image from '../components/LoginImage/LoginImage';
import AuthLayout from '../components/AuthLayout/AuthLayout';
  

const Login = (props) => {
  return (
        <AuthLayout
        text={
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vesti"
        }
        left={
          <LoginForm />
        } 
        right={
          <Image />
        }
        />
    );
}

export default Login
