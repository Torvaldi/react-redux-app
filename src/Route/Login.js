import React from 'react';

import LoginForm from '../containers/form/LoginForm';
import BlockText from '../components/BlockText';
import Image from '../components/LoginImage';

import '../css/loginRoute.css';

const Login = (props) => {
  return (
        <section className="authLayout" >
          <article className="authBlock">
            <BlockText />
            <LoginForm />
          </article>
          <article className="imageLayout">
            <Image />
          </article>
        </section>
    );
}

export default Login
