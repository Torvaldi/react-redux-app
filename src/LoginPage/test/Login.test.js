import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Login from '../Login';

configure({ adapter: new Adapter() });

describe('<Login />', () => {
    it('should render Login Componenent', () => {
        const wrapper = shallow(<Login />);
        expect(wrapper).toMatchSnapshot();
    });
});