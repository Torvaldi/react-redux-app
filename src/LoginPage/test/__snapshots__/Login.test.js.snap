// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render Login Componenent 1`] = `ShallowWrapper {}`;
